fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -56,39 +56,39 @@ def __init__(self, X, sensitive_features, y, estimator, constraints, eps, B, opt
         self.last_linprog_n_hs = 0
         self.last_linprog_result = None
 
-    def _eval_from_error_gamma(self, error, gamma, lambda_vec):
+    def _eval(self, h, lambda_vec):
         """Return the value of the Lagrangian.
 
-        :return: tuple `(L, L_high)` where `L` is the value of the Lagrangian, and
-            `L_high` is the value of the Lagrangian under the best response of the lambda player
-        :rtype: tuple of two floats
+        :param h: TODO: should this be Q instead of h? It seems to indicate how much weight each h
```

Comment:
@MiroDudik see the question in the code:
:param h: TODO: should this be Q instead of h? It seems to indicate how much weight each h gets in the randomized classifier, summing up to 1

Reply:
Correct. The name of the argument should be Q, and the type should be either: Series with weights that sum to 1, or a callable (corresponding to a *deterministic* predict).

Reply:
Great, adjusted that